openedx_learning/apps/core/itemstore/models.py

```diff
@@ -0,0 +1,133 @@
+"""
+The item model hiearachy is: Item -> ItemVersion -> ItemInfo -> ItemRaw
+
+Item is the versionless thing that is guaranteed to exist for the lifetime of
+the LearningContext. An ItemVersion is a different version of that item for a
+given LearningContext, and may include policy changes (like grading). ItemInfo
+covers basic metadata that is intrinsic to the item itself, and now how it's
+used in a LearningContext. ItemRaw represents the raw byte data.
+
+TODO: Add link to ADR after it merges.
+"""
+from django.db import models
+
+from openedx_learning.lib.fields import (
+    hash_field,
+    identifier_field,
+    immutable_uuid_field,
+    manual_date_time_field,
+)
+from ..publish.models import LearningContext, LearningContextVersion
+
+
+class Item(models.Model):
+    """
+    This represents any content that has ever existed in a LearningContext.
+    """
+    uuid = immutable_uuid_field()
+    learning_context = models.ForeignKey(LearningContext, on_delete=models.CASCADE)
+    identifier = identifier_field()
+
+    # TODO: These pointers to the latest published version are convenient, but
+    # we don't currently have the data integrity guarantees to make sure that 
+    # multiple versions aren't active at the same time. Maybe add a ref to the
+    # Item model in LearningContextVersionItemVersion, so we can force that
+    # constraint?
+    published_version = models.ForeignKey('ItemVersion', on_delete=models.RESTRICT, null=True, related_name="+")
+    first_published_at = models.DateTimeField(null=True)  # can be null if never published 
+    last_published_at = models.DateTimeField(null=True)   # can be null if never published
+
+    created = manual_date_time_field()
+    modified = manual_date_time_field()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["learning_context_id", "identifier"],
+                name="learning_publishing_lcb_one_identifier_per_lc",
+            )
+        ]
+
+    def __str__(self):
+        return f"{self.identifier}"
+
+
+class ItemVersion(models.Model):
+    """
+    A particular version of an Item.
+
+    A new ItemVersion is created anytime there is either a change to the content
+    or a change to the policy around a piece of content (e.g. schedule change).
+    """
+    uuid = immutable_uuid_field()
+    item_info = models.ForeignKey('ItemInfo', on_delete=models.RESTRICT)
+    item = models.ForeignKey(Item, on_delete=models.CASCADE)
+    title = models.CharField(max_length=1000, blank=True, null=True)
```

Comment:
Why is `title` here and not in ItemInfo? Is it because it's example of policy, which we're distinguishing from metadata (like `type`)?

Reply:
Yes. Renaming is desirable for libraries, because you don't know what kind of convention that library is using, e.g. "Problem 22a" may not make as much sense for the new context it's being used in.

Reply:
Makes sense!

Reply:
Should this be `title_override` then to clarify that you can optionally stick with whatever the library author gave as a title?

Reply:
I'm inclined to not use the `_override` convention because the default content in the library would also be using this field (just under the content library's learning context).

Though maybe the title should be lifted out into a separate model? Maybe even one with language awareness?

Wait... maybe ItemRaw:ItemVersion is M:1?

I'll make comments along both of those lines for future followup, but won't make any immediate model changes.